### PR TITLE
Remove chai-as-promised and convert tests to async/await

### PR DESCRIPTION
**Pull request by dstaley**
In preparation for running tests in an actual browser environment, I've refactored all the tests that relied on `chai-as-promised` to async/await. In the process I discovered several tests that weren't waiting on promises correctly, and were throwing unhandled rejections.

Removing `chai-as-promised` was necessary to get the tests passing in [Intern](https://theintern.io/). I'll send a future PR actually adding Intern, but I figured this would be a good PR to merge even without Intern.

### CODE REVIEW

**Comment by dstaley** [comment]
@leplatrem would love a quick spot check on this! For the most part it's exactly the same, so just skimming the changes and checking a few spots would be really appreciated!

File: test/bucket_test.ts [hunk 1]
```diff
@@ -42,26 +44,26 @@ describe("Bucket", () => {
...
 
-    it("should resolve with response data", () => {
+    it("should resolve with response data", async () => {
       const response = { data: { foo: "bar" }, permissions: {} };
       sandbox.stub(client, "execute").returns(Promise.resolve(response));
 
-      return getBlogBucket()
-        .getData()
-        .should.become({ foo: "bar" });
+      ((await getBlogBucket().getData()) as { foo: string }).should.deep.equal({
```
Comment (leplatrem) [inline test/bucket_test.ts]: Maybe in this case, an intermediary variable would make sense

File: test/integration_test.ts [hunk 2]
```diff
@@ -1274,151 +1185,136 @@ describe("Integration tests", function() {
...
-            .then(({ data }) => {
-              expect(data[0].title).eql("foo");
-              expect(data[0].members).eql(["github:me"]);
-              expect(data[0].blah).eql(43);
-            });
+        it("should update a group", async () => {
+          const res = await bucket.createGroup("foo");
+          await bucket.updateGroup({ ...res.data, title: "mod" });
+          const { data } = await bucket.listGroups();
+          (data[0] as Group & { title: string }).title.should.equal("mod");
```
Comment (leplatrem) [inline test/integration_test.ts]: Oh, i don't know this syntax :grin: ! Could you add a comment please?

File: test/bucket_test.ts (hunk 1 above)
Comment (dstaley) [reply test/bucket_test.ts]: I've updated the tests to use an intermediary variable whenever we were casting an awaited value. However, we still have tests that look like this:

```ts
(await api.fetchServerInfo()).should.deep.equal(fakeServerInfo);
```

Is this clear enough, or should we lift all awaited calls into variables?

Comment (leplatrem) [reply test/bucket_test.ts]: Yes, this was like one of the few complex ones. Thanks for the changes!

**Pull request merged**